### PR TITLE
Fix flaking flat strings and add Checksums

### PR DESCRIPTION
**Pull request by puerco**
This PR fixes the flat strings used to serialize the various elements for comparison and checksumming. The first iteration was running into flakiness when some of the secondary structures were being serialized using their memory address. This PR fixes that and adds a lot more tests.

This PR is needed ahead of the conformance test framework which I'll PR once this fix merges.

Signed-off-by: Adolfo Garcia Veytia (puerco) <puerco@chainguard.dev>

### CODE REVIEW

**Comment by jspeed-meyers** [comment]
@puerco -- Both the linter and the tests are failing. Any thoughts on what's going on? I'll review anyway.

File: pkg/sbom/node.go [hunk 1]
```diff
@@ -211,14 +213,76 @@ func (n *Node) Equal(n2 *Node) bool {
...
+			for _, i := range n.Suppliers {
+				pairs = append(pairs, fmt.Sprintf("supplier:%s", i.flatString()))
+			}
+		case "bomsquad.protobom.Node.originators":
+			for _, i := range n.Originators {
+				pairs = append(pairs, fmt.Sprintf("originator:%s", i.flatString()))
+			}
+		case "bomsquad.protobom.Node.identifiers":
+			for _, i := range n.Identifiers {
+				pairs = append(pairs, i.flatString())
```
Comment (jspeed-meyers) [inline pkg/sbom/node.go:231]: Is the GitHub linter grumpy because the `flatString()` function is recursive?

File: pkg/sbom/node_test.go [hunk 2]
```diff
@@ -184,3 +185,97 @@ func TestGetRootNodes(t *testing.T) {
 		require.Equal(t, tc.expected, nodes)
 	}
 }
+
+func TestNodeFlatString(t *testing.T) {
```
Comment (jspeed-meyers) [inline pkg/sbom/node_test.go:189]: Nice tests!

**Review by jspeed-meyers — COMMENTED** [review]
Code looks good. But linter and tests are grumpy....

File: pkg/sbom/node.go (hunk 1 above)
Comment (puerco) [reply pkg/sbom/node.go:231]: I had forgotten to check in a file. Fixed now!

**Review by jspeed-meyers — APPROVED** [review]
LGTM

File: pkg/sbom/identififer.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package sbom
```
[the remark this answers is not in the document]
Comment (puerco) [reply pkg/sbom/identififer.go]: Sorry, I need to work with my glasses on :upside_down_face: 
Done

**Pull request merged**